### ISSUE TITLE
Missing `return` in read_heroku

### ISSUE THREAD

**Issue by dgouldin** [pganalyze/pganalyze-collector#7]
`read_heroku` never actually returns `config`:

https://github.com/pganalyze/pganalyze-collector/blob/master/pgacollector/Configuration.py#L22-L34

**Comment by dgouldin** [comment]
Actually, looks like there's more than that wrong. `api_key` and `api_url` should also be included in the config that gets returned by `read_heroku`

**Comment by lfittl** [comment]
@dgouldin Yeah, that was a rather hack-ish approach to configuration handling - agreed that those are wrong.

There is actually a bigger refactoring/rewrite going on at https://github.com/pganalyze/collector which is zooming in on a stable release in the next 1-2 weeks. Essentially turning it from a cron-based Python script to a proper Go daemon.

Its more likely we'll add Heroku support there in a clean way, rather than fixing this soon deprecated repository. Is this a blocking issue for you to use pganalyze, or were you just curious about the script?

**Comment by dgouldin** [comment]
We use envdir via encrypted chef data bags for our EC2 config (along with the url-style db configuration), so I read the source and added the necessary env vars to fool the script into using the heroku approach. I don't particularly care about heroku compatibility, but it'd be nice to have the option for env var based config rather than having my db auth duplicated in a separate file somewhere.

**Comment by lfittl** [comment]
@dgouldin Ah, that makes a ton of sense of course!

In the new approach there is full support for a `DB_URL` environment variable (regardless of whether you're monitoring self-hosted/RDS/Heroku), so this should be fixed then.

On that, I'm actually curious - as mentioned the collector will now actually be a daemon instead of a simple script, hence there is some additional support needed (sysvinit/upstart/etc script). We'll try to ship packages for all common (Linux) distributions and init systems.

Anything we can do to make your life easier in upgrading to that new daemon-based approach, in terms of packaging? (also curious if there is any easy-to-handle projects/products you have in your environment)
